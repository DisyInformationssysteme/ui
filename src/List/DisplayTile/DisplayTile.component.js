import React from 'react';

/**
 * @param {object} props react props
 * @example
<DisplayTile name="Hello world"></DisplayTile>
 */
function DisplayTile(props) {
	return (
		<div className="panel panel-default">
			<pre>{JSON.stringify(props, null, 2)}</pre>
		</div>
	);
}

DisplayTile.propTypes = {
	items: React.PropTypes.arrayOf(
		React.PropTypes.object
	),
	columns: React.PropTypes.arrayOf(
		React.PropTypes.object
	),
};

export default DisplayTile;
