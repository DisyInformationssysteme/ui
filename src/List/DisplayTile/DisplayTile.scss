// DisplayTile styles

/// tile background color
/// @type Color
$tile-background: #FFF;
/// tile border color
/// @type Color
$tile-border-color: #CCC;
/// all tile title color
/// @type Color
$title-color: #555;
/// Global font size for tile rendering
/// @type Number (Unit)
$text-size: 15px;

.tiles {
	display: flex;
	flex-wrap: wrap;
	justify-content: flex-start;
	list-style: none;
	padding: 0 20px;
}

.tile {
	background: $tile-background;
	border: 1px solid $tile-border-color;
	border-radius: 0.5em;
	font-size: $text-size;
	height: 250px;
	margin: 0 25px 25px 0;
	padding: 15px;
	transition: 0.13s background cubic-bezier(0.18, 0.89, 0.32, 1.28), 0.13s border-color cubic-bezier(0.18, 0.89, 0.32, 1.28);

	&:hover {
		background: lighten($title-color, 60%);
		border-color: darken($tile-border-color, 40%);
	}
}

.title {
	color: $title-color;
	font-size: $text-size;
	font-weight: 600;
	letter-spacing: 0;
	margin: 0;
	text-transform: uppercase;
}

.titlelink {
	padding: 0;

	&:hover {
		text-decoration: underline;
	}
}

.itemlist {
	padding-left: 6em;

	&:after {
		content: '';
		clear: both;
		display: table;
	}
}

.itemtitle,
.itemvalue {
	float: left;

	&:last-of-type {
		margin-bottom: 0;
	}
}

.itemtitle {
	color: $title-color;
	clear: left;
	font-size: 0.8em;
	font-weight: 400;
	line-height: 2.5em;
	margin-left: -7.5em;
	text-align: right;
	text-transform: uppercase;
	width: 7em;

	&:after {
		content: ' :';
	}
}

.itemvalue {
	clear: right;
	line-height: 2em;
	margin-left: 0; // reset default
	overflow: hidden;
	text-overflow: ellipsis;
	width: 9em;
	white-space: nowrap;
}
