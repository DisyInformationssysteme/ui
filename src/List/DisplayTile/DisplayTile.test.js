import React from 'react';
import renderer from 'react-test-renderer';

import DisplayTile from './DisplayTile.component';

describe('DisplayTile', () => {
	it('should render its name', () => {
		const wrapper = renderer.create(<DisplayTile name="Hello world" />).toJSON();
		expect(wrapper).toMatchSnapshot();
	});
});
