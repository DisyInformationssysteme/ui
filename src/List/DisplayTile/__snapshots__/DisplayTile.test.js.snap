exports[`DisplayTile should render its name 1`] = `
<div
  className="panel panel-default">
  <pre>
    {
      \"name\": \"Hello world\"
    }
  </pre>
</div>
`;
